components/Budget/CreateBudgetModal.jsx

```diff
@@ -1,42 +1,100 @@
+                    participantArray.forEach((participant) => {
+                        participantObject[participant] = false;
+                    });
+                    setParticipants(participantObject);
+                    setLoading(False);
+                }
+            );
+        } catch (error) {
+            console.log("uh oh: ", error.message);
+        }
+    }, []);
 
     const createBudget = () => {
-        // set(ref(db, `productions/${productionCode}/budgets`), {
-        //     [budgetID]: Date.now(),
-        // });
-        // set(ref(db, "budgets/" + budgetID), {
-        //     name: name,
-        //     budget: budget,
-        // });
-        // closeModal();
-        alert("Budget created!");
+        if (name === "" || budget === "") {
+            setAlert(
+                "Please fill in all fields.",
+                "bg-red-500",
+                icon({ name: "circle-exclamation" })
+            );
+            return;
+        }
+
+        const participantsToAdd = {};
+        for (const participant in participants) {
+            if (participants[participant]) {
+                participantsToAdd[participant] = Date.now();
+            }
+        }
+
+        const budgetID = Math.floor(Math.random() * 10 ** 20);
+        try {
+            set(ref(db, `productions/${productionCode}/budgets`), {
+                [budgetID]: Date.now(),
+            });
+            set(ref(db, "budgets/" + budgetID), {
+                name: name,
+                budget: budget,
+                participants: participantsToAdd,
+            });
+            setAlert("Budget created!", "bg-green-500", icon({ name: "circle-exclamation" }));
+            setModal(null);
+        } catch {
+            setAlert(
+                "Error occurred when creating budget.",
+                "bg-red-500",
+                icon({ name: "circle-exclamation" })
+            );
+        }
     };
 
     return (
         <View className="flex flex-col p-3 h-full">
             <Text className="text-3xl font-extrabold text-center mb-3">Create Budget</Text>
-            <View className="flex-1" />
-            <Text className="text-lg font-semibold text-center">
-                Enter the name for your budget.
-            </Text>
+            <Text className="text-lg font-semibold text-center">Name</Text>
             <FormField value={name} placeholder="Name" onChangeText={setName} />
-            <Text className="text-lg font-semibold text-center">
-                Enter the budget for your budget.
-            </Text>
-            <FormField
-                value={budget}
-                placeholder="Budget"
-                onChangeText={setBudget}
-            />
-            <View className="flex-1" />
+            <View className="h-2" />
+            <Text className="text-lg font-semibold text-center">Budget (£)</Text>
+            <FormField value={budget} placeholder="Budget" onChangeText={setBudget} />
```

Comment:
add a check for only numbers here

Reply:
It's in submit form

Reply:
it is clearly not working very well unfortunately 
<img width="367" alt="image" src="https://github.com/DanWilkins2107/TheatreApp/assets/147443062/993347c1-3e1d-490f-9652-b0c157c30c9a">

Reply:
I'd say regex in the setter is the best option like how I didn't allow spaces in email addresses

Reply:
`onChangeText={(text) => setEmail(text.replace(/\s/g, ""))}`

Reply:
Don't think it's needed? Due to isNan setting the values?